### ISSUE TITLE
failure to push remote tf state when not using default region

### ISSUE THREAD

**Issue by davemeyer** [quintilesims/layer0#444]
### Expected behavior
tf state should be pushed to s3 bucket in the region specified in l0-setup init

### Actual behavior
the state is pushed to a nonexistent bucket in the default refion (us-west-2), so no remote state is ever saved.

[INFO] Pushing /Users/dmeyer/.layer0/hexeast1/dockercfg.json to s3://layer0-hexeast1-236817160978/terraform/dockercfg.json
BucketRegionError: incorrect region, the bucket is not in 'us-west-2' region

v0.10.3

**Comment by jparsons04** [comment]
Do you happen to have the environment variable `LAYER0_AWS_REGION` set in your environment? I ask because `l0-setup push` invokes `newAWSProviderHelper` which in turn checks for the cli flag `aws-region`.

```12:52:08 jparsons@seawl000002 {(v0.10.3)} ~/dev/go/src/github.com/quintilesims/layer0/setup $ ./l0-setup push --help
NAME:
   l0-setup push - push a Layer0 instance configuration to S3

USAGE:
   l0-setup push [command options] NAME

OPTIONS:
   --aws-access-key value  access key portion of an AWS key [$LAYER0_AWS_ACCESS_KEY_ID]
   --aws-secret-key value  secret key portion on an AWS key [$LAYER0_AWS_SECRET_ACCESS_KEY]
   --aws-region value      AWS region [$LAYER0_AWS_REGION]
```
If you aren't passing in `--aws-region` as a flag for the command `l0-setup push` and you don't have the environment variable `LAYER0_AWS_REGION` set, then it will default to us-west-2:
https://github.com/quintilesims/layer0/blob/v0.10.3/setup/command/aws.go#L58

**Comment by jparsons04** [comment]
The PR that I just merged into `develop` would require `LAYER0_AWS_REGION` to be set in the environment. If you don't have this and run `l0-setup` then you need to pass in the `--aws-region` flag. There is no default region any longer, and it needs to be explicitly provided.

In my testing, I used both the environment variable and cli flag, and created and applied a new layer0 instance in us-east-1 and was able to get the terraform state pushed to the S3 bucket housed in us-east-1 without issue.

**Comment by jparsons04** [comment]
Just want to leave a comment with a summary of our upcoming changes:

1. The l0-setup cli flag `--aws-region` has been removed as an option.
2. The region you specify in `l0-setup init` means that all of the layer0 instance's resources and the S3 bucket will be created in that region.
3. The `l0-setup` commands have been fixed to be able to correctly calculate what region the bucket is in.

I tested this code in us-west-2, us-east-1 and eu-west-1. In each case, I was able to `l0-setup init` and `l0-setup apply` and each run successfully saved my instance's configuration and state to the S3 bucket. `l0-setup pull` and `l0-setup push` were also tested in these regions and worked correctly.
